client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx

```diff
@@ -1,260 +1,238 @@
 import OtherPublicLocationForm from './PlacesAdditionalForms/OtherPublicLocationForm';
 import MedicalLocationForm from './PlacesAdditionalForms/MedicalLocationForm';
+import useInteractionsForm from './useInteractionsForm';
+import InteractionEventDialogFields from '../InteractionsEventDialogContext/InteractionEventDialogFields';
 
 export const defaultContact: Contact = {
   firstName: "",
   lastName: "",
-  phoneNumber: { number: "", isValid: true },
+  phoneNumber: "",
   id: "",
-  contactType: -1
+  contactType: -1,
 };
 
-const addContactButton: string = 'הוסף מגע';
-
-const InteractionEventForm: React.FC<Props> = (props: Props) : JSX.Element => {
-    const {
-        interactionEventDialogData,
-        setInteractionEventDialogData,
-    } = useContext(InteractionEventDialogContext);
-    const {
-        placeType,
-        startTime,
-        endTime,
-        externalizationApproval,
-        contacts,
-        placeSubType,
-        id,
-        investigationId,
-    } = interactionEventDialogData;
-
-    const classes = useStyles();
-    const formClasses = useFormStyles();
-    const { setDefaultPlaceName } = props;
-    const [canAddContact, setCanAddContact] = useState<boolean>(false);
-
-    const {
-        geriatric,
-        school,
-        medical,
-        office,
-        otherPublicPlaces,
-        privateHouse,
-        religion,
-        transportation,
-    } = placeTypesCodesHierarchy;
-
-    React.useEffect(() => {
-        const hasInvalidContact: boolean = contacts.some(
-            (contact) =>
-                !contact.firstName || !contact.lastName || !contact.phoneNumber
-        );
-        setCanAddContact(!hasInvalidContact);
-    }, [contacts]);
-
-    const onContactAdd = () => {
-        const updatedContacts = [...contacts, {...defaultContact}];
-        setInteractionEventDialogData({
-            ...interactionEventDialogData,
-            contacts: updatedContacts,
-        });
-    };
-
-    const onPlaceTypeChange = (newPlaceType: string) => {
-        setInteractionEventDialogData({
-            ...initialDialogData(startTime, endTime, contacts, investigationId),
-            id,
-            placeType: newPlaceType,
-            externalizationApproval,
-        });
-    };
-
-    const placeNameByPlaceType = (placeSubTypeDispalyName?: string) => {
-        if (placeType === transportation.code && placeSubTypeDispalyName) {
-            return `${placeType} ${placeSubTypeDispalyName}`
-        }
-        return undefined;
+const addContactButton: string = "הוסף מגע";
+
+const InteractionEventForm: React.FC<Props> = (
+  { intractionData,
+    loadInteractions,
+    closeNewDialog,
+    closeEditDialog,
+    interactionId 
+  } : Props): JSX.Element => {
+
+    const { schema, saveIntreactions } = useInteractionsForm({ interactionId, loadInteractions, closeNewDialog, closeEditDialog });
+    const methods = useForm<InteractionEventDialogData>({
```

Comment:
why not destructing the methods?

Reply:
I used react-hook-form context, when I call useFormContext as you can see in Contact Form for example I use destructing